File: app/llms/solid/javascript/lsp.js


class Bird {
  fly() {
    console.log("Flying");
  }
}


class Ostrich extends Bird {
  fly() {
    throw new Error("Ostrich cannot fly");
  }
}

class FlyingBird {
  fly() {}
}

class Sparrow extends FlyingBird {
  fly() {
    console.log("Flying");
  }
}

class Ostrich {}

File: app/llms/solid/golang/lsp.go
package main

type Bird interface {
	Fly()
}

type Ostrich struct{}

func (o Ostrich) Fly() {
	panic("Ostrich cannot fly")
}

type FlyingBird interface {
	Fly()
}

type Sparrow struct{}

func (s Sparrow) Fly() {}
